rust/cardano-blockchain-types/src/txn_index.rs

```diff
@@ -21,3 +21,80 @@ impl TxnIndex {
         Self(value)
     }
 }
+
+impl<
+        T: Copy
+            + TryInto<u16>
+            + std::ops::Sub<Output = T>
+            + std::cmp::PartialOrd<T>
+            + num_traits::identities::Zero,
+    > From<T> for TxnIndex
+{
+    fn from(value: T) -> Self {
+        TxnIndex::from_saturating(value)
```

Comment:
Should be able to just use `from_saturating` in the `conversion.rs` crate directly, don't need the `from_saturating` method otherwise, as this will do the same.